src/coordinator/storage/local/storage_test.go

```diff
@@ -123,16 +223,101 @@ func TestLocalRead(t *testing.T) {
 	assert.Equal(t, tags, results.SeriesList[0].Tags)
 }
 
-func setupLocalSearch(t *testing.T) storage.Storage {
+func TestLocalSearchError(t *testing.T) {
 	ctrl := gomock.NewController(t)
-	store, session := setup(ctrl)
-	session.EXPECT().FetchTaggedIDs(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, false, errors.ErrNotImplemented)
-	return store
-}
+	defer ctrl.Finish()
+	store, sessions := setup(t, ctrl)
+	sessions.forEach(func(session *client.MockSession) {
+		session.EXPECT().FetchTaggedIDs(gomock.Any(), gomock.Any(), gomock.Any()).
+			Return(nil, false, fmt.Errorf("an error"))
+	})
 
-func TestLocalSearchExpectedFail(t *testing.T) {
-	store := setupLocalSearch(t)
 	searchReq := newFetchReq()
 	_, err := store.FetchTags(context.TODO(), searchReq, &storage.FetchOptions{Limit: 100})
 	assert.Error(t, err)
 }
+
+func TestLocalSearchSuccess(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store, sessions := setup(t, ctrl)
+
+	type testFetchTaggedID struct {
+		id        string
+		namespace string
+		tagName   string
+		tagValue  string
+	}
+
+	fetches := []testFetchTaggedID{
+		{
+			id:        "foo",
+			namespace: "metrics_unaggregated",
+			tagName:   "qux",
+			tagValue:  "qaz",
+		},
+		{
+			id:        "bar",
+			namespace: "metrics_aggregated",
+			tagName:   "qel",
+			tagValue:  "quz",
+		},
+	}
+
+	sessions.forEach(func(session *client.MockSession) {
+		var f testFetchTaggedID
+		switch {
+		case session == sessions.unaggregated1MonthRetention:
+			f = fetches[0]
+		case session == sessions.aggregated1MonthRetention1MinuteResolution:
+			f = fetches[1]
+		default:
+			panic("unexpected session")
```

Comment:
Might be better to use `require.FailNow` to avoid a panic?

Reply:
Sure thing.